widget/contrib/wifi.lua

```diff
@@ -0,0 +1,70 @@
+local awful     = require("awful")
+local wibox     = require("wibox")
+local helpers   = require("lain.helpers")
+
+function factory(args)
+    local wifi      =  { widget = wibox.widget.textbox() }
+    local args      = args or {}
+    local timeout   = args.timeout or 2
+    local settings  = args.settings or function() end
+         
+    local command = {
+        check_conn = "[ ! -z \"`ping -c1 -w3 www.duckduckgo.com`\" ] && echo -n 1",
+        nmcli_dev = "nmcli -t dev | grep -E ^wl",
+        nmcli_wifi = "nmcli -g IN-USE,SSID,SIGNAL dev wifi list | grep -F \"*\"",
+    }
+
+    wifi_now = {
+        device = args.device or "N/A",
+        connected = false,
+        ssid = args.ssid_placeholder or "N/A",
+        signal = args.signal_placeholder or "N/A",
+    }
+
+    function wifi.shell_cmd(cmd)
+        return awful.util.shell .. " -c '" .. cmd .. "'"
+    end
+
+    function wifi.update()
+
+        awful.spawn.easy_async(wifi.shell_cmd(command.check_conn),
```

Comment:
Use instead [this](https://awesomewm.org/doc/api/libraries/awful.spawn.html#easy_async_with_shell) function I pushed a while ago.